tree/dataframe/inc/ROOT/RDF/RLoopManager.hxx

```diff
@@ -118,7 +118,7 @@ class RLoopManager : public RNodeBase {
    std::shared_ptr<TTree> fTree{nullptr};
    const ColumnNames_t fDefaultColumns;
    ULong64_t fStartEntry{0};
-   ULong64_t fEndEntry{0};
+   ULong64_t fEndEntry{std::numeric_limits<ULong64_t>::max()};
```

Comment:
The only new data member should be `fDatasetSpec`. Also these two need to go

Reply:
This was our initial idea, but I realized it also makes sense to keep these 2 attributes.

Reason why I kept `fStartEntry` and `fEndEntry` is that the RLoopManager constructor accepts an `RDatasetSpec` as an argument and handles all other specifications inside of the constructor already.

So the internal chain is already handled, and so are the friends. Why would I need to store strings of file and tree names, when I have already constructed everything?

Remark is that, even if I provide a spec as a member, I still need a vector of TTrees to hold the friends.

Reply:
I'm on the fence about this. on the one hand, it avoids all the issues that we were discussing about creating a spec, passing it to RDF, then changing the spec by adding friends. In the docs, we can say that the information about the dataset is only handled in the constructor then forgotten. On the other hand, maybe there are cases where we want to have quick access to the list of file names passed by the user. Currently, we would need to retrieve them from the internal TChain. By keeping the RDatasetSpec as data member we could just get the filenames from it.

Reply:
hmm, I am thinking about it. I think I can come up with valid counter arguments:

I tend to agree with the first argument - especially that we should put that clearly in the docs. But I think whether the spec is a method or not, this snippet would be "equally bad". Right?
```
spec = RDatasetSpec("tree", "file")
df = RDataFrame(spec)
spec.AddFriend("fr1", "file1")
```

And regarding retrieving tree and file names: Hmm, the potential use case, sounds reasonable to me, but I see complications there. I am afraid that having the spec as a method might lead to more confusion. Because if the user want to retrieve this information, they would be able to, only if they constructed the RLoopManager with RDatasetSpec. Otherwise, we would need to adapt existing methods to also update the spec method of the RLoopManager.
I think it suffices to have public getters inside of the RDatasetSpec which fetch that info upon request. User only needs to recall which spec belonged to their dataframe.